www/class/CentreonLDAPAdmin.class.php

```diff
@@ -167,30 +194,37 @@ public function setGeneralOptions(array $options, $arId = 0)
                     "LDAP PARAM - Warning the reference date wasn\'t set for LDAP : " . $options['ar_name']
                 );
             }
-            $this->db->query(
-                "INSERT INTO auth_ressource (ar_name, ar_description, ar_type, ar_enable, ar_sync_base_date) 
-                VALUES ('" . $this->db->escape($options['ar_name']) . "',
-                    '" . $this->db->escape($options['ar_description']) . "',
-                    'ldap',
-                    '" . $options['ldap_auth_enable']['ldap_auth_enable'] . "',
-                    '" . $options['ar_sync_base_date'] . "')"
+            $statement = $this->db->prepare(
+                "INSERT INTO auth_ressource (ar_name, ar_description, ar_type, ar_enable, ar_sync_base_date)
+                VALUES (:ar_name, :ar_description,'ldap', :ar_enable, :ar_sync_base_date)"
             );
-            $maxArIdSql = "SELECT MAX(ar_id) as last_id
-                          FROM auth_ressource
-                          WHERE ar_name = '" . $this->db->escape($options['ar_name']) . "'";
-            $res = $this->db->query($maxArIdSql);
+            $statement->bindValue(':ar_name', $options['ar_name'], \PDO::PARAM_STR);
```

Comment:
shouldn't $options['ar_name'] be also sanitized ?

Reply:
it's already sanitized on form.php line 339 ($value used on setGeneralOption called in line 395)